src/pages/createTask.jsx

```diff
@@ -0,0 +1,179 @@
+import React, { useMemo, useState } from 'react'
+import { Link, useLocation, useNavigate } from 'react-router-dom'
+
+import styled from 'styled-components'
+import { colors } from 'shared/constants/colors'
+
+import NavBar from 'shared/components/NavBar'
+import Text from 'shared/components/Text'
+import Button from 'shared/components/Button'
+import Input from 'shared/components/Input'
+
+const BUTTON_TEXT = Object.freeze({
+	VALID: '계속 진행하기',
+	INVALID: '할 일을 입력해주세요',
+})
+
+export const CreateTask = () => {
+	const navigate = useNavigate()
+	const location = useLocation()
+
+	const [spareTime, setSpareTime] = useState({ hour: 0, minute: 0 })
+	const [task, setTask] = useState('')
+	const [tasks, setTasks] = useState([])
+	const [isValidTasks, setIsValidTasks] = useState(false)
+
+	const handleSubmit = e => {
+		e.preventDefault()
+		const trimmedTask = task.trim()
+
+		if (trimmedTask) {
+			setTasks([...tasks, { id: `${Date.now()}${tasks.length}`, task: trimmedTask }])
+		}
+
+		setTask('')
+	}
+
+	const removeTask = removeId => {
+		const filteredTasks = tasks.filter(({ id }) => removeId !== id)
+		setTasks(filteredTasks)
+	}
+
+	const handleIsValidTask = tasks => {
+		if (tasks.length > 0) {
+			setIsValidTasks(true)
+			return
+		}
+		setIsValidTasks(false)
+	}
+
+	useMemo(() => {
+		try {
+			const { spareTime } = location.state
+			setSpareTime(spareTime)
+		} catch {
+			navigate('/home')
+		}
+	}, [location, navigate])
+
+	useMemo(() => {
+		handleIsValidTask(tasks)
+	}, [tasks])
+
+	return (
+		<Wrapper>
+			<NavBar backIcon />
+
+			<SubTitle>
+				<Text style={{ textAlign: 'start', fontSize: '2.2rem', padding: '0 3rem' }}>
+					오늘 해야할 일들은 무엇이 있나요?
+				</Text>
+				<Text
+					style={{ textAlign: 'start', fontSize: '1.3rem', color: '#919191', padding: '0 3rem' }}
+				>
+					클릭하여 삭제할 수 있습니다.
+				</Text>
+			</SubTitle>
+
+			<Section>
+				<TaskArea>
+					{tasks.map(({ id, task }) => (
+						<Task key={id}>
+							<span onClick={() => removeTask(id)}>{task}</span>
+						</Task>
+					))}
+				</TaskArea>
+				<Form onSubmit={e => handleSubmit(e)}>
```

Comment:
오 button onClick으로 하지 않고 Form으로 한 것이 인상적이네요!! (나쁜 뜻 아님)


P5;
받아올 params와 넘겨줄 params가 같을 경우 아래와 같이 생략도 가능합니다.

```suggestion
				<Form onSubmit={handleSubmit(}>
```